### PR TITLE
Add black configuration

### PR DESCRIPTION
**Pull request by sayanchowdhury**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[tool.black]
```
Comment (ananyo2012) [inline pyproject.toml:1]: Can you move this to `setup.cfg` ? That's the place all the pre-commit configs are present. I think black config should go there as well.

**Comment by ananyo2012** [comment]
Also is this work overlapping woth #667 ?

**Comment by sayanchowdhury** [comment]
No, this is not overlapping with the #667 work.

File: pyproject.toml (hunk 1 above)
Comment (sayanchowdhury) [reply pyproject.toml:1]: I don't think `poetry` recognizes `setup.cfg` file

**Comment by ananyo2012** [comment]
@sayanchowdhury Why do we need poetry ?

**Comment by sayanchowdhury** [comment]
> @sayanchowdhury Why do we need poetry ?

`pyproject.toml` is the new preferred way to package the project and poetry eases the use of it.

File: pyproject.toml (hunk 1 above)
Comment (pradyunsg) [reply pyproject.toml:1]: It does not, no.

**Comment by pradyunsg** [comment]
It's not a "poetry configuration" file. This was standardized in PEP 518, which introduced the pyproject.toml file format. poetry builds upon this standard, utilizing the "space" within that file that it has access to.

ttps://www.python.org/dev/peps/pep-0518/#tool-table

**Comment by sayanchowdhury** [comment]
> It's not a "poetry configuration" file. The `pyproject.toml` file was standardized in PEP 518. poetry and black, utilize the "space" within that file that they are allowed to use (`tool.pypi-name`).
> 
> https://www.python.org/dev/peps/pep-0518/#tool-table

Right, that was the mistake on my part. That would not be poetry configuration rather the PEP.  And going forward I plan to integrate poetry, and also possibly move everything from `setup.cfg` to `pyproject.toml`.  I know there is an open ticket in `isort` to support this, which @pradyunsg created long back.

**Comment by ananyo2012** [comment]
If we are going for a dependency manager I would rather prefer pipenv.

**Comment by sayanchowdhury** [comment]
As of now this PR is for black configuration. If we have to migrate to another dependency manager we can have a discussion and let work accordingly

**Comment by palnabarun** [comment]
@sayanchowdhury Please squash the commits/rebase your branch to remove the merge commit.

**Comment by palnabarun** [comment]
Oops! Closing was a misclick.

**Comment by sayanchowdhury** [comment]
Closing in favour of #677

**Pull request closed without merging**